src/compiler/binder.ts

```diff
@@ -3495,6 +3495,9 @@ namespace ts {
             case SyntaxKind.BreakStatement:
                 transformFlags |= TransformFlags.ContainsHoistedDeclarationOrCompletion;
                 break;
+
+            case SyntaxKind.ImportCallExpression:

```

Comment:
Now that I'm back on a computer, I found the discussion in estree's repository about the AST node: https://github.com/estree/estree/pull/137

And this is how it got merged in their spec: https://github.com/estree/estree/blob/master/experimental/import-expression.md

Thinking about it, (when it gets meta-properties) it'll behave pretty similarly to `super`, being both callable and "indexable".